cohortextractor2/dsl.py

```diff
@@ -158,12 +165,149 @@ class PatientSeries:
     variable.
     """
 
-    def __init__(self, value: Value):
+    def __init__(self, value: Value | Comparator):
         self.value = value
 
+    def is_comparator(self) -> bool:
```

Comment:
It would be good to mark this as private by prefixing its name with an underscore. We need to be really careful about the surface we expose to study authors -- they shouldn't need to know this exists.

Reply:
will do